src/effects/SoundTouchEffect.cpp

```diff
@@ -110,43 +110,41 @@ bool EffectSoundTouch::ProcessWithTimeWarper(InitFunction initer,
             bGoodResult = false;
       }; },
 #endif
-      [&](auto &&fallthrough){ return [&](WaveTrack &leftTrack) {
-         if (!leftTrack.GetSelected())
+      [&](auto &&fallthrough){ return [&](WaveTrack &orig) {
+         if (!orig.GetSelected())
             return fallthrough();
 
          // Process only if the right marker is to the right of the left marker
          if (mT1 > mT0) {
+            //Transform the marker timepoints to samples
+            const auto start = orig.TimeToLongSamples(mT0);
+            const auto end = orig.TimeToLongSamples(mT1);
+
+            const auto tempList = orig.WideEmptyCopy();
+            auto &out = **tempList->Any<WaveTrack>().begin();
+
             const auto pSoundTouch = std::make_unique<soundtouch::SoundTouch>();
             initer(pSoundTouch.get());
 
             // TODO: more-than-two-channels
-            auto channels = TrackList::Channels(&leftTrack);
-            auto rightTrack = (channels.size() > 1)
-               ? * ++ channels.first
-               : nullptr;
-            if (rightTrack) {
-               //Transform the marker timepoints to samples
-               auto start = leftTrack.TimeToLongSamples(mT0);
-               auto end = leftTrack.TimeToLongSamples(mT1);
+            auto channels = orig.Channels();
+            if (channels.size() > 1) {
 
                //Inform soundtouch there's 2 channels
                pSoundTouch->setChannels(2);
 
                //ProcessStereo() (implemented below) processes a stereo track
                if (!ProcessStereo(pSoundTouch.get(),
-                  &leftTrack, rightTrack, start, end, warper))
+                  orig, out, start, end, warper))
```

Comment:
There is post processing to restore clip boundaries, remapped by the time stretch.

The logic for this is unfortunately replicated in SBSMSEffect, Nyquist, and EQ rather than having a unified implementation applied to all effects.

I became aware of this problem in other refactoring but decided not to address it.  The scope of the restructuring is already big enough.